packages/tools/src/tools/annotation/ContourROITool.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  getEnabledElement,
+  Types,
+  utilities as csUtils,
+  StackViewport,
+  VolumeViewport,
+} from '@cornerstonejs/core';
+import registerContourDrawLoop from './contourROITool/drawLoop';
+import { EventTypes, PublicToolProps, ToolProps } from '../../types';
+import PlanarFreehandROITool from './PlanarFreehandROITool';
+import { getViewportIdsWithToolToRender } from '../../utilities/viewportFilters';
+import triggerAnnotationRenderForViewportIds from '../../utilities/triggerAnnotationRenderForViewportIds';
+import InterpolationManager from '../../utilities/contourROITool/InterpolationManager';
+import { InterpolationROIAnnotation } from '../../types/ToolSpecificAnnotationTypes';
+import * as annotation from '../../stateManagement/annotation';
+
+const { getImageSliceDataForVolumeViewport } = csUtils;
+
+class ContourROITool extends PlanarFreehandROITool {
+  constructor(toolProps: PublicToolProps = {}, defaultToolProps: ToolProps) {
+    super(toolProps, defaultToolProps);
+    this.configuration.allowOpenContours = false;
+    this.configuration.interpolation.interpolateOnAdd = false;
+    this.configuration.interpolation.interpolateOnEdit = false;
+    this.configuration.makeClockWise = true;
+    InterpolationManager.addTool(ContourROITool.toolName);
+
+    // Register event loops and rendering logic, which are stored in different
+    // Files due to their complexity/size.
+    registerContourDrawLoop(this);
+  }
+
+  getSliceData(viewport, referencedImageId): Types.ImageSliceData {
```

Comment:
Need docs here - I was thinking you were returning image pixel values, but what you are actually returning is number of slices etc.  I'd suggest an external utilities function that gets this data.  I had calling it slice data, but that is kind of what it is.
Maybe just enhance getImageSliceDataForVolumeViewport with something that handles Stack as well, call it getImageIndexSlices
Then, more implementations can be added to your new location.